src/signer/mod.rs

```diff
@@ -1,45 +1,38 @@
-use crate::{error::Error, keys::Key, prefix::SeedPrefix};
-use ed25519_dalek::{Keypair, PublicKey, SecretKey};
+use crate::{error::Error, keys::{PrivateKey, PublicKey}, prefix::{SeedPrefix}};
+use ed25519_dalek::{Keypair, SecretKey};
 use rand::rngs::OsRng;
 pub trait KeyManager {
-    fn sign(&self, msg: &Vec<u8>) -> Result<Vec<u8>, Error>;
-    fn public_key(&self) -> Key;
-    fn next_public_key(&self) -> Key;
+    fn sign(&self, msg: &[u8]) -> Result<Vec<u8>, Error>;
+    fn public_key(&self) -> PublicKey;
+    fn next_public_key(&self) -> PublicKey;
     fn rotate(&mut self) -> Result<(), Error>;
 }
 
 pub struct CryptoBox {
     signer: Signer,
-    next_priv_key: Key,
-    pub next_pub_key: Key,
-    seeds: Vec<String>,
+    next_priv_key: PrivateKey,
+    pub next_pub_key: PublicKey,
+    seeds: Box<dyn Iterator<Item = SeedPrefix>>,
```

Comment:
Why Box`ing here instead of CryptoBox<T> where T: Iterator<Item = SeedPrefix> ?

Reply:
Because empty iterator is needed here: https://github.com/decentralized-identity/keriox/blob/39f222f0d1645e5a616249fe04ccbff3d316a3b0/src/signer/mod.rs#L57
and I couldn't do this that way.

Reply:
the question is why trait objects (runtime cost) versus generic (compiling time cost)?

Reply:
this one is still not resolved, right? =)

Reply:
It is. There is no seeds iterator now, so no trait objects/generic :D

Reply:
perfect. thanks. you can merge it now =)